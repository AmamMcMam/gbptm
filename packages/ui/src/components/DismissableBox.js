import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from './css/dismissable-box.module.css';

import config from '../config.js';

class DismissableBox extends Component {
  constructor(props) {
    super(props);

    // Set the initial state
    this.state = {
      dimissed: false,
    };

    this.onDismiss = this.onDismiss.bind(this);
  }

  onDismiss() {
    var persistKey = this.props.persistKey;

    this.setState({
      dismissed: true,
    });

    if (persistKey) {
      config.setSetting('dismissed', persistKey, true);
    }

    this.props.onDismiss();
  }

  render() {
    var persistKey = this.props.persistKey;

    if (
      this.state.dismissed ||
      (persistKey && config.getSetting('dismissed', persistKey))
    ) {
      return null;
    }

    return (
      <div className={styles.container}>
        <h2 className={styles.title}>{this.props.title}</h2>
        <div>{this.props.content}</div>
        <button className={styles.close} onClick={this.onDismiss}>
          âœ–
        </button>
      </div>
    );
  }
}

DismissableBox.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,

  // If specified, the dismissed state will be persisted with this identifier
  // in a browser cookie
  persistKey: PropTypes.string,

  // Callback fn triggered once a message has been dismissed
  onDismiss: PropTypes.func,
};

DismissableBox.defaultProps = {
  onDismiss: Function.prototype,
};

export default DismissableBox;
