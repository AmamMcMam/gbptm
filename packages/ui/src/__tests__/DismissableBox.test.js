import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import DismissableBox from '../components/DismissableBox';

var title = 'Lorem Ipsum';
var content = '<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>';

// Jest snapshot
it('renders snapshot', () => {
  var tree = renderer
    .create(<DismissableBox title={title} content={content} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('persists', () => {
  var component = mount(
    <DismissableBox title={title} content={content} persistKey="persist" />
  );

  // Click close button
  component.find('button').simulate('click');

  // Ensure persistance object is set on localStorage
  expect(localStorage.setItem).toBeCalledWith(
    'dismissed',
    JSON.stringify({ persist: true })
  );
});
