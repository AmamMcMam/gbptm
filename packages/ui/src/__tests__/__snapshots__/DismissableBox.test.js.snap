// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<div
  className="container"
>
  <h2
    className="title"
  >
    Lorem Ipsum
  </h2>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>",
      }
    }
  />
  <button
    className="close"
    onClick={[Function]}
  >
    âœ–
  </button>
</div>
`;
